tests/llm_cache.py

```diff
@@ -267,25 +269,10 @@ def get_engine(
             worker.model_runner.n_blocks = available_blocks
             # need to overwrite the block pool and kv cache manager if the
             # number of available blocks has changed
-            kwargs = {
-                "hash_block_size": SpyrePlatform.get_block_size()
-            } if has_argument(BlockPool, "hash_block_size") else {}
-            worker.model_runner.block_pool = BlockPool(
-                num_gpu_blocks=available_blocks + 1,
-                enable_caching=use_pc,
-                enable_kv_cache_events=False,
-                **kwargs)
-            worker.model_runner.kv_cache_manager = FullAttentionManager(
-                kv_cache_spec=worker.model_runner._attn_spec,
-                block_pool=worker.model_runner.block_pool,
-                # Currently don't support models with more than one
-                # attention type, e.g. full and sliding window, so
-                # there is only one group.
-                kv_cache_group_id=0,
-                # We don't support DCP
-                # https://docs.vllm.ai/en/latest/serving/context_parallel_deployment/#decode-context-parallel
-                dcp_world_size=1,
-            )
+            worker.model_runner.block_pool = (
+                worker.model_runner._make_block_pool())
+            worker.model_runner.kv_cache_manager = (
+                worker.model_runner._make_kv_cache_manager())
```

Comment:
hell yeah 👍 

I wouldn't mind a `worker.model_runner.reset_kv_cache()` but this is waaaaay better already